dev-utils/run-script.js

```diff
@@ -158,21 +160,28 @@ function runSauceTests(packagePath, serve = 'true', ...scripts) {
   })
 }
 
-let cleanUps = []
-function exitHandler(exitCode) {
-  console.log('Running cleanups.')
-  cleanUps.forEach(f => {
-    try {
-      f(exitCode)
-    } catch (e) {
-      console.error(e)
+function exitHandler(callExit, exitCode) {
+  if (cleanUps.length > 0) {
+    console.log('Running cleanups:', cleanUps.length)
+    cleanUps.forEach(f => {
+      try {
+        f(exitCode)
+      } catch (e) {
+        console.error(e)
+      }
+    })
+    cleanUps = []
+  }
+  if (callExit) {
+    if (exitCode !== 0) {
+      console.log('Exiting with code:', exitCode)
     }
-  })
-  cleanUps = []
+    process.exit(exitCode)
```

Comment:
we might miss the console.log above since its async and this exit would call synchronously. Its better to set the exitcode instead of doing thiss
```js
process.exitCode = exitCode
```